src/main/java/com/hackclub/hccore/advancements/MusicophileAdv.java

```diff
@@ -0,0 +1,79 @@
+import java.util.List;
+
+public class MusicophileAdv extends BaseAdvancement {
+    static AdvancementDisplayBuilder<AdvancementDisplay.Builder, AdvancementDisplay> displayBuilder =
+            new AdvancementDisplay.Builder(Material.JUKEBOX, "Musicophile")
+                    .challengeFrame()
+                    .announceChat()
+                    .showToast()
+                    .description("Collect every single music disc");
+    static Material[] musicDiscs = {Material.MUSIC_DISC_13, Material.MUSIC_DISC_CAT, Material.MUSIC_DISC_BLOCKS,
+            Material.MUSIC_DISC_CHIRP, Material.MUSIC_DISC_FAR, Material.MUSIC_DISC_MALL, Material.MUSIC_DISC_MELLOHI
+            , Material.MUSIC_DISC_STAL, Material.MUSIC_DISC_STRAD, Material.MUSIC_DISC_WARD, Material.MUSIC_DISC_11,
+            Material.MUSIC_DISC_WAIT, Material.MUSIC_DISC_OTHERSIDE, Material.MUSIC_DISC_5,
+            Material.MUSIC_DISC_PIGSTEP};
+    @SuppressWarnings("unchecked")
+    public MusicophileAdv(HCCorePlugin plugin, Advancement root, AdvancementKey key, CoordAdapter adapter) {
+        super(key.getKey(), displayBuilder.coords(adapter, key).build(), root, musicDiscs.length);
+
+        for (Material disc : musicDiscs) {
+            registerEvent(InventoryClickEvent.class, e -> {
+                if (e.getCurrentItem().getType() == disc) {
+                    HumanEntity entity = e.getWhoClicked();
+
+                    List<String> metadataDiscs;
+                    if (entity.hasMetadata("hccore_discs")) {
+                        metadataDiscs = (List<String>)entity.getMetadata("hccore_discs").get(0).value();
+                    } else {
+                        metadataDiscs = new ArrayList<>();
+                    }
+
+                    if (metadataDiscs.contains(disc.toString())) return;
+                    metadataDiscs.add(disc.toString());
+
+                    entity.setMetadata("hccore_discs",
+                           new FixedMetadataValue(plugin, metadataDiscs));
+                    incrementProgression(e.getWhoClicked().getUniqueId());
+                }
+            });
+
+            registerEvent(EntityPickupItemEvent.class, e -> {
+                if (e.getItem().getItemStack().getType() == disc) {
+                    LivingEntity entity = e.getEntity();
+
+                    List<String> metadataDiscs;
+                    if (entity.hasMetadata("hccore_discs")) {
+                        metadataDiscs = (List<String>)entity.getMetadata("hccore_discs").get(0).value();
+                    } else {
+                        metadataDiscs = new ArrayList<>();
+                    }
+
+                    if (metadataDiscs.contains(disc.toString())) return;
+                    metadataDiscs.add(disc.toString());
+
+                    entity.setMetadata("hccore_discs",
+                            new FixedMetadataValue(plugin, metadataDiscs));
+
+                    incrementProgression(e.getEntity().getUniqueId());
```

Comment:
this could be a non-player

Reply:
EntityPickupItemEvent#getEntity returns living entity, so there is not always a player behind the uuid

Reply:
additionally, incrementProgression wants a player object, not a uuid